### ISSUE TITLE
login.js fails with 'Could not generate "googleAuth" object.'

### ISSUE THREAD

**Issue by demonbane** [Brandawg93/homebridge-nest-cam#303]
Running homebridge on a Raspberri Pi, I'm getting a "could not generate googleauth object" error when I try to run login.js. This has been working fine for me for a few months already, but just stopped working today.

I'm running login with `homebridge-nest-cam login -p /usr/bin/chromium-browser -h`. The entire browser workflow goes as usual with me logging in without a problem. After logging in the browser closes, and my terminal shows the error where I used to get the credentials printed. I've also tried running with `node dist/login.js` from the module directory and still get the same issue.

Everything is up to date according to Homebridge Config UI X.

**Comment by Brandawg93** [comment]
The latest version of the plugin (v5.4.2) should be more descriptive on what went wrong during the login process. Let me know if this helps.

**Comment by mzaur** [comment]
I have the same issue on windows, and I'm on homebridge-nest-cam v5.5.0. If I try through the UI it says 

"Could not generate "googleAuth" object. You can still manually retrieve the issue token and enter it below."

If I try in command prompt it says:

Could not generate "googleAuth" object.

Below are the values retrieved by the login process:

issueToken: undefined,
cookies: undefined

I'm not sure what I am doing wrong. I just installed Homebridge today but did follow all the instructions.

**Comment by Brandawg93** [comment]
@mzaur Are you able to get your login credentials using the manual web scraping method? The UI should be doing the exact same thing but it seems to be having issues for some.

**Comment by mzaur** [comment]
> @mzaur Are you able to get your login credentials using the manual web scraping method? The UI should be doing the exact same thing but it seems to be having issues for some.

I tried that with Chrome but it didn't seem to work out. I followed the instructions [here](https://github.com/Brandawg93/homebridge-nest-cam/wiki/Manual-Authentication) and was on step 6. I am not sure what 'iframerpc' call means. But I have two "issue_jwt" under headers, and neither one has a URL beginning with accounts.google.com.
Instead the two Request URLs are both https://nestauthproxyservice-pa.googleapis.com/v1/issue_jwt

I did clear all cookies too. Did Google change something?

**Comment by Brandawg93** [comment]
That usually means that your browser is holding onto your credentials. Try a new browser or using incognito mode.

**Comment by mzaur** [comment]
Ok I was able to get it. I forgot to click 'preserve log'
Do I have to do this every time I re-auth? Do you need any logs or anything to figure out why the normal login method doesn't work?

**Comment by Brandawg93** [comment]
I just published v5.5.1 that should fix this issue. If someone has a chance to confirm this works, please let me know.

**Comment by demonbane** [comment]
Just tested 5.5.1 on the Raspberry Pi and now it is providing the `cookies` value but not `issueToken`:
```
pi@homebridge:~ $ homebridge-nest-cam login -h
Opening chromium browser...
Could not generate "googleAuth" object. 

Below are the values retrieved by the login process:

issueToken: undefined,
cookies: [REDACTED]
```

Using the manual method as described in the wiki still seems to work ok.

**Comment by demonbane** [comment]
Just noticed that the automated method returns a lot more data for the cookies than what I get with the manual method. These are the keys that are included from `homebridge-nest-cam login -h`:
```
pi@homebridge:~ $ tr ';' '\n' < testlogin | cut -f1 -d'='
user_id
 NID
 SIDCC
 __Host-GAPS
 ACCOUNT_CHOOSER
 __Secure-3PSIDCC
 SSID
 __Secure-3PAPISID
 SMSV
 __Secure-3PSID
 APISID
 LSID
 SAPISID
 SID
 __Host-3PLSID
 HSID
```

And here's what I get from the manual method:
```
pi@homebridge:~ $ tr ';' '\n' < testdevtools | cut -f1 -d'='
__Secure-3PSID
 __Host-3PLSID
 __Secure-3PAPISID
 NID
 __Secure-3PSIDCC
```

Not sure if that's relevant or not, just thought it was worth pointing out.

**Comment by Brandawg93** [comment]
@demonbane That's good info. Thanks! Does it work for you without the `-h`? I was experiencing the same issue when trying it in headful mode.

**Comment by Brandawg93** [comment]
I just published a new test version that should fix this issue. You can try it out by installing the latest test version via config-ui or via `npm -g i homebridge-nest-cam@5.5.2-test.0 --unsafe-perm`. Let me know if this works for you.

**Comment by demonbane** [comment]
Looks good! Just tried 5.5.2-test (still in headful mode since I'm using 2FA), and it worked without a problem.

**Comment by Brandawg93** [comment]
Great to hear! I've published v5.5.2 which includes these changes.
